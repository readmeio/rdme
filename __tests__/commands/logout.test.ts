import { describe, afterEach, beforeAll, it, expect } from 'vitest';

import pkg from '../../package.json';
import Command from '../../src/commands/logout.js';
import configStore from '../../src/lib/configstore.js';
import { runCommand } from '../helpers/setup-oclif-config.js';

describe('rdme logout', () => {
  let run: (args?: string[]) => Promise<string>;

  beforeAll(() => {
    run = runCommand(Command);
  });

  afterEach(() => {
    configStore.clear();
  });

  it("should report the user as logged out if they aren't logged in", () => {
    configStore.delete('email');
    configStore.delete('project');

    return expect(run()).resolves.toBe(
      `You have logged out of ReadMe. Please use \`${pkg.name} login\` to login again.`,
    );
  });

  it('should log the user out', async () => {
    configStore.set('email', 'email@example.com');
    configStore.set('project', 'subdomain');

    await expect(run()).resolves.toBe(
      `You have logged out of ReadMe. Please use \`${pkg.name} login\` to login again.`,
    );

    expect(configStore.get('email')).toBeUndefined();
    expect(configStore.get('project')).toBeUndefined();
  });
});
