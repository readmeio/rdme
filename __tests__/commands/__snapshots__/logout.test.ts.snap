// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rdme logout > should log the user out 1`] = `
{
  "result": "You have logged out of ReadMe. Please use \`rdme login\` to login again.",
  "stderr": "",
  "stdout": "",
}
`;

exports[`rdme logout > should report the user as logged out if they aren't logged in 1`] = `
{
  "result": "You have logged out of ReadMe. Please use \`rdme login\` to login again.",
  "stderr": "",
  "stdout": "",
}
`;
